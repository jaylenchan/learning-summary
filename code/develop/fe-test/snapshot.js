export function genConfig() {
  return {
    server: 'http://localhost',
    port: 80
  }
}

export function genConfig1() {
  return {
    domain: 'www.baidu.com'
  }
}

export function genConfigWithTime() {
  return {
    time: new Date()
  }
}
