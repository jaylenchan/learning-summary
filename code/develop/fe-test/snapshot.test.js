import { genConfig, genConfig1, genConfigWithTime } from './snapshot'

/**
 * 快照测试的一个应用场景是：配置文件的更改测试。比如一个函数会生成一个配置文件
 */
describe('', () => {
  test('', () => {
    expect(genConfig()).toMatchSnapshot()
  })

  test('', () => {
    expect(genConfig1()).toMatchSnapshot()
  })

  test('', () => {
    expect(genConfigWithTime()).toMatchSnapshot({
      time: expect.any(Date)
    })
  })

  test('行内的snapshot', () => {
    expect(genConfigWithTime()).toMatchInlineSnapshot(
      {
        time: expect.any(Date)
      },
      `
      Object {
        "time": Any<Date>,
      }
    `
    )
  })
})
