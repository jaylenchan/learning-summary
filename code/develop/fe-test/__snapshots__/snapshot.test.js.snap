// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "port": 80,
  "server": "http://localhost",
}
`;

exports[` 2`] = `
Object {
  "domain": "www.baidu.com",
}
`;

exports[` 3`] = `
Object {
  "time": Any<Date>,
}
`;

exports[` 4`] = `
Object {
  "time": Any<Date>,
}
`;
